src/SFA.DAS.Authorization.CommitmentPermissions/CommitmentPermissionsAuthorizationRegistry.cs

```diff
@@ -1,12 +1,26 @@
-﻿using StructureMap;
+﻿using SFA.DAS.Authorization.Cache;
+using SFA.DAS.Authorization.CommitmentPermissions.Cache;
+using StructureMap;
 
 namespace SFA.DAS.Authorization.CommitmentPermissions
 {
     public class CommitmentPermissionsAuthorizationRegistry : Registry
     {
         public CommitmentPermissionsAuthorizationRegistry()
         {
-            For<IAuthorizationHandler>().Add<AuthorizationHandler>();
+            For<IAuthorizationHandler>().Add<AuthorizationHandler>()
+                .DecorateWith((ctx, handler) => BuildCache(ctx, handler))
```

Comment:
Could the adapter registration also be added to `ServiceCollectionExtensions`? Here's an example of how to do this with `IServiceCollection` if you need one: https://github.com/SkillsFundingAgency/das-authorization/blob/master/src/SFA.DAS.Authorization/ServiceCollectionExtensions.cs#L15